gettsim/generic_functions.py

```diff
@@ -38,31 +38,41 @@ def fill_intercepts_at_lower_thresholds(
 def get_dict_of_arrays_piecewise_linear(list_of_dicts):
     """Extract the relevant parameters for e_anr_frei.
 
-    Args:
-        list_of_dicts: list_of_dicts specifying upper_thresholds and
-              respective rates for a given legislature time.
-
-    Returns:
-        dict of 3 arrays for calc_e_anr_frei: upper_thresholds, rates and
-        intercepts_at_lower_thresholds.
-    """
-    upper_thresholds = np.array([d["upper_threshold"] for d in list_of_dicts],
-                                dtype=np.float)
-    rates = np.array([d["rate"] for d in list_of_dicts], dtype=np.float)
-    intercepts = fill_intercepts_at_lower_thresholds(
-        upper_thresholds, rates, 0, piecewise_linear
-    )
+        Args:
+            list_of_dicts: list_of_dicts specifying upper_thresholds and
+                  respective rates for a given legislature time.
+
+        Returns:
+            dict of 3 arrays for calc_e_anr_frei: upper_thresholds, rates and
+            intercepts_at_lower_thresholds.
+        """
+    keys_without_reference = list(list_of_dicts.keys())
+    keys_without_reference.remove("reference")
```

Comment:
Sorry, my bad. Should have been a bit more verbose here. 

In the syntax of the YAML-file, I tried to be as close to a list as possible while maintaining the possibility to reference an element (see the examples below your main usecase; for kids or for the 2011 change). 

Let us please take the opposite route (picking the keys we want rather than discarding the irrelevant ones). So something like the following should do:

 ```python
keys = sorted([key for key in list_of_dicts.keys() if type(key) == int])
```